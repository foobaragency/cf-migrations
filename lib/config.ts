export const config = {
  contentful: {
    defaultLocale: "en-US",
    contentType: {
      name: "Migrations",
      id: "migrations",
    },
  },
} as const
