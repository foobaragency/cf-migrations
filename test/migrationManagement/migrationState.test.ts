import {
  generateMigrationStates,
  getPendingMigrationFilePaths,
} from "../../lib/migrationManagement/migrationState"
import { config } from "../config"

describe("Migration State", () => {
  it("generates a valid migration state update based on the pending migrations", () => {
    const pendingMigrations = ["0001-migration", "0002-migration"]
    const migrationStateUpdatePayload = generateMigrationStates(
      pendingMigrations
    )

    expect(migrationStateUpdatePayload).toMatchSnapshot()
  })

  it("filters the pending migrations", () => {
    const deployedMigrations = ["0001-migration", "0002-migration"]
    const migrationFileNames = [...deployedMigrations, "0003-migration"]

    const { pendingMigrations } = getPendingMigrationFilePaths(
      config.migrationsDirectory,
      deployedMigrations,
      migrationFileNames
    )

    expect(pendingMigrations).toEqual(["0003-migration"])
  })
})
