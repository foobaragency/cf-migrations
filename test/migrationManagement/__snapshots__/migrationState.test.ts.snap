// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migration State generates a valid migration state update based on the pending migrations 1`] = `
Array [
  Object {
    "fileName": Object {
      "en-US": "0001-migration",
    },
  },
  Object {
    "fileName": Object {
      "en-US": "0002-migration",
    },
  },
]
`;
